libeos-parental-controls/tests/app-filter.c

```diff
@@ -236,6 +289,972 @@ test_app_filter_builder_copy_full (void)
+                                                 const gchar            *property_name)
+{
+  if (g_str_equal (property_name, "AppFilter"))
+    return data->expected_app_filter_value;
+  else if (g_str_equal (property_name, "OarsFilter"))
+    return data->expected_oars_filter_value;
+  else if (g_str_equal (property_name, "AllowUserInstallation"))
+    return data->expected_allow_user_installation_value;
+  else if (g_str_equal (property_name, "AllowSystemInstallation"))
+    return data->expected_allow_system_installation_value;
+  else
+    g_assert_not_reached ();
+}
+
+/* This is run in a worker thread. */
+static void
+set_app_filter_server_cb (GtDBusQueue *queue,
+                          gpointer     user_data)
+{
+  const SetAppFilterData *data = user_data;
+  g_autoptr(GDBusMethodInvocation) find_invocation = NULL;
+  g_autofree gchar *object_path = NULL;
+  g_autofree gchar *find_reply = NULL;
+  g_autoptr(GVariant) properties_variant = NULL;
+
+  g_assert ((data->error_index == -1) == (data->dbus_error_name == NULL));
+  g_assert ((data->dbus_error_name == NULL) == (data->dbus_error_message == NULL));
+
+  /* Handle the FindUserById() call. */
+  gint64 user_id;
+  find_invocation =
+      gt_dbus_queue_assert_pop_message (queue,
+                                        "/org/freedesktop/Accounts",
+                                        "org.freedesktop.Accounts",
+                                        "FindUserById", "(x)", &user_id);
+  g_assert_cmpint (user_id, ==, data->expected_uid);
+
+  object_path = g_strdup_printf ("/org/freedesktop/Accounts/User%u", (uid_t) user_id);
+  find_reply = g_strdup_printf ("(@o '%s',)", object_path);
+  g_dbus_method_invocation_return_value (find_invocation, g_variant_new_parsed (find_reply));
+
+  /* Handle the Properties.Set() calls. */
+  const gchar *expected_properties[] =
+    {
+      "AppFilter",
+      "OarsFilter",
+      "AllowUserInstallation",
+      "AllowSystemInstallation",
+    };
+  gsize i;
+
+  for (i = 0; i < G_N_ELEMENTS (expected_properties); i++)
+    {
+      const gchar *property_interface;
+      const gchar *property_name;
+      g_autoptr(GVariant) property_value = NULL;
+      g_autoptr(GDBusMethodInvocation) property_invocation = NULL;
+      g_autoptr(GVariant) expected_property_value = NULL;
+
+      property_invocation =
+          gt_dbus_queue_assert_pop_message (queue,
+                                            object_path,
+                                            "org.freedesktop.DBus.Properties",
+                                            "Set", "(&s&sv)", &property_interface,
+                                            &property_name, &property_value);
+      g_assert_cmpstr (property_interface, ==, "com.endlessm.ParentalControls.AppFilter");
+      g_assert_cmpstr (property_name, ==, expected_properties[i]);
+
+      if (data->error_index >= 0 && (gsize) data->error_index == i)
```

Comment:
I feel like making `i` have the same type as `data->error_index` would be neater than needing this cast. (Do you even need this cast to a wider type, actually?)

I know that conceptually, an array index should be of type `size_t`, but in this case, we know its largest value is `4`.

Reply:
Fair enough.